src/pages/DataResearch.tsx

```diff
@@ -0,0 +1,227 @@
+import React, { useMemo } from 'react'
+import { useGroupBy } from 'src/api/groupByService'
+import Widget from 'src/shared/Widget'
+import { useDate } from './components/DateTimePicker'
+import moment from 'moment'
+import { Grid } from '@mui/material'
+import { DateSelector } from './components/DateSelector'
+import { useTranslation } from 'react-i18next'
+import { Skeleton } from 'antd'
+import {
+  Area,
+  Tooltip,
+  AreaChart,
+  CartesianGrid,
+  ResponsiveContainer,
+  XAxis,
+  YAxis,
+} from 'recharts'
+import { PageContainer } from './components/PageContainer'
+import { getColorName } from './dashboard/AllLineschart/OperatorHbarChart/OperatorHbarChart'
+
+const now = moment()
+const unique: (value: string, index: number, self: string[]) => boolean = (value, index, self) =>
+  self.indexOf(value) === index
+
+export const DataResearch = () => {
+  return (
+    <PageContainer>
+      <Widget>
+        <h1>מחקרים</h1>
+        <p>אם יש לכם רעיון מעניין למה קורים פה דברים, דברו איתנו בסלאק!</p>
+      </Widget>
+      <StackedResearchSection />
+    </PageContainer>
+  )
+}
+
+function StackedResearchSection() {
+  const [startDate, setStartDate] = useDate(now.clone().subtract(7, 'days'))
+  const [endDate, setEndDate] = useDate(now.clone().subtract(1, 'day'))
+  const [groupByHour, setGroupByHour] = React.useState<boolean>(false)
+  const [graphData, loadingGraph] = useGroupBy({
+    dateTo: endDate,
+    dateFrom: startDate,
+    groupBy: groupByHour ? 'operator_ref,gtfs_route_hour' : 'operator_ref,gtfs_route_date',
+  })
+
+  return (
+    <Widget>
+      <h1>בעיות etl/gps/משהו גלובאלי אחר</h1>
+      <StackedResearchInputs
+        startDate={startDate}
+        setStartDate={setStartDate}
+        endDate={endDate}
+        setEndDate={setEndDate}
+        groupByHour={groupByHour}
+        setGroupByHour={setGroupByHour}
+      />
+      <StackedResearchChart
+        graphData={graphData}
+        isLoading={loadingGraph}
+        field="total_actual_rides"
+        title="מספר נסיעות בפועל"
+        description="כמה נסיעות נרשמו כהתבצעו בכל יום/שעה בטווח הזמן שבחרתם. (נסיעות = siri rides)"
+      />
+      <StackedResearchChart
+        graphData={graphData}
+        isLoading={loadingGraph}
+        field="total_planned_rides"
+        title="מספר נסיעות מתוכננות"
+        description="כמה נסיעות היו אמורות להיות בכל יום/שעה בטווח הזמן שבחרתם. (נסיעות = נסיעות מתוכננות בgtfs)"
+      />
+      <StackedResearchChart
+        graphData={graphData}
+        isLoading={loadingGraph}
+        field="total_missed_rides"
+        title="מספר נסיעות שלא התבצעו"
+        description="כמה נסיעות היו אמורות להיות בכל יום/שעה בטווח הזמן שבחרתם אבל לא התבצעו. (הפרש בין שני הגרפים הקודמים)"
+      />
+    </Widget>
+  )
+}
+
+function StackedResearchInputs({
```

Comment:
You should move it above the StackedResearchSection component

Reply:
Why?

Reply:
Hmm, you don't have to, but when I read the file from above to down you use functions and then declare them. However, you use the component in different file so it is not give an errors, up to you.

Reply:
Usually, I prefer putting helper function and small utilities at the bottom, so when you read it you'll see the important things first.
But it's just an opinion
https://softwareengineering.stackexchange.com/questions/196997/helper-methods-placement

Reply:
Ok 👍